server/fishtest/util.py

```diff
@@ -63,20 +64,28 @@ def get_chi2(tasks, bad_users):
                 float(stats.get("draws", 0)),
             ]
         else:
-            p = stats["pentanomial"]
+            p = stats.get("pentanomial", 5 * [0])  # there was a small window
+            # in time where we could have both trinomial and pentanomial
+            # workers
+
             # The ww and ll frequencies will typically be too small for
             # the full pentanomial chi2 test to be valid. See e.g. the last page of
             # https://www.open.ac.uk/socialsciences/spsstutorial/files/tutorials/chi-square.pdf.
             # So we combine the ww and ll frequencies with the wd and ld frequencies.
             wld = [float(p[4] + p[3]), float(p[0] + p[1]), float(p[2])]
-        if wld == [0.0, 0.0, 0.0]:
-            continue
         if key in users:
             for idx in range(len(wld)):
                 users[key][idx] += wld[idx]
         else:
             users[key] = wld
 
+    users_ = copy.deepcopy(users)
+    for key in users_:
+        if min(users[key]) < 10:  # the normal rule is expected >= 5
+            # but this is inconvenient since we haven't computed the
+            # expected frequencies at this point
+            del users[key]
+
```

Comment:
IMO this filtering is helpful If we want only to use chi2 only like a meter.
This filtering is problematic if we want to use chi2 to auto purge a tests because (if I read correctly the code) in this way we don't compute the residuals for the low count workers, keeping all the low count workers results (low total of course).
In a test with few workers at high concurrency and many workers at very low concurrency with low cell count, the former will dominate the expected cell count (so under H0 they will have low residuals and low chi2 contribution), the latter could have high residuals lowering/polluting the p-value (if used like a meter). But in that case the low concurrency workers are the candidates to be purged if the p-value is less than the critical one.

Reply:
I don't think that low count workers need to have bad data. Also they don't have to have low core count. They may be high core count workers that have been stopped prematurely. 

I have been looking at earlier tests and with the new book there are very few workers that don't have enough games to be included in the chi2 test, even at LTC.

I don't think a standardized residual for a worker with low cell count makes a lot of sense since the distribution is not normal. Once could try to be more accurate but since such workers contribute little to the outcome of the test I think it is not worth it.